File: Easy/Reach a given score.cpp
//{ Driver Code Starts
// Driver Code
#include<bits/stdc++.h>
#define ll long long int
using namespace std;

// } Driver Code Ends
// Complete this function



long long int count(long long int n)
{
	long long int table[n+1],i;
	memset(table, 0, sizeof(table));
	table[0]=1;                 // If 0 sum is required number of ways is 1.
	
	// Your code here
	for (int i=0; i<3; i++)
	{
	    for (int j=0; j<=n; j++)
	    {
	        if      (i==0 && j-3 >=0)  table[j] += table[j-3];
	        else if (i==1 && j-5 >=0)  table[j] += table[j-5];
	        else if (i==2 && j-10 >=0) table[j] += table[j-10];
	    }
	}
	
	
	return table[n];
}

//{ Driver Code Starts.



int main()
{
	int t;
	cin>>t;
	while(t--)
	{
		ll n;
		cin>>n;
		cout<<count(n)<<endl;
	}
	return 0;
}
// } Driver Code Ends

File: Easy/Reach a given score.py
#User function Template for python3

class Solution:
    def count(self, n: int) -> int:
        #your code here
        dp = [0] * (n + 1)
        dp[0] = 1
        
        for i in range(3):
            for j in range(n+1):
                if i == 0 and j-3 >= 0:
                    dp[j] += dp[j-3]
                elif i == 1 and j-5 >= 0:
                    dp[j] += dp[j-5]
                elif i == 2 and j-10 >= 0:
                    dp[j] += dp[j-10]
        
        return dp[n]
        
        


#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == "__main__":
    for _ in range(int(input())):
        n = int(input())
        ob = Solution()
        print(ob.count(n))
        
# } Driver Code Ends